### PR TITLE
Refactor Project Files

### PR DESCRIPTION
**Pull request by wsugarman**
## Description
- Update project dependencies
    - Ensure all direct dependencies are listed explicitly
    - Bump versions
- Add package descriptions
- Fix CI build warnings from SourceLink concerning missing submodules

## Related issues
N/A

## Testing
Pass unit tests

### CODE REVIEW

File: src/Microsoft.Health.Dicom.SqlServer.UnitTests/Microsoft.Health.Dicom.SqlServer.UnitTests.csproj [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
     <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3" />
   </ItemGroup>
+
   <ItemGroup>
+    <ProjectReference Include="..\Microsoft.Health.Dicom.Core\Microsoft.Health.Dicom.Core.csproj" />
     <ProjectReference Include="..\Microsoft.Health.Dicom.SqlServer\Microsoft.Health.Dicom.SqlServer.csproj" />
-    <ProjectReference Include="..\Microsoft.Health.Dicom.Tests.Common\Microsoft.Health.Dicom.Tests.Common.csproj" />
-  </ItemGroup>
-  <ItemGroup>
-    <Folder Include="Features\Query\" />
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.SqlServer.UnitTests/Microsoft.Health.Dicom.SqlServer.UnitTests.csproj:17]: This is included already in the new build SDK

**Review by KnicKnic — APPROVED** [review]

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core.UnitTests/Microsoft.Health.DicomCast.Core.UnitTests.csproj [hunk 2]
```diff
@@ -1,15 +1,30 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
+
   <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
   </PropertyGroup>
+
   <ItemGroup>
+    <PackageReference Include="fo-dicom" Version="4.0.7" />
+    <PackageReference Include="Hl7.Fhir.R4" Version="1.9.0" />
```
Comment (brandonpollett) [inline converter/dicom-cast/src/Microsoft.Health.DicomCast.Core.UnitTests/Microsoft.Health.DicomCast.Core.UnitTests.csproj:9]: If we're explicitly referencing them, we most likely need to update our third party references file.

File: global.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
-    "sdk": {
-        "version": "5.0.103"
-    }
+  "sdk": {
+    "version": "5.0.103",
+    "rollForward": "latestMajor"
```
Comment (brandonpollett) [inline global.json:4]: I know in the past, we've run into breaking behaviors in patch versions. That's generally why we require specific versions to force the check with what we expect to be running on.

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Microsoft.Health.Dicom.Web.Tests.E2E.csproj [hunk 4]
```diff
@@ -1,86 +1,61 @@
...
-  <Target Name="VerifyExactSdkVersion" BeforeTargets="Build">
-    <!--
-      Verify that the we are compiling with the exact version of the SDK that is specified in the global.json file.
-      If the version specified in the file is not installed on the system, dotnet uses the latest installed version instead.
-      We want to avoid that behavior because new SDK versions by default target the runtime version it is released with,
-      so the runtime behavior could be different depending on what SDK versions were installed on the machine that compiled
-      the code.
-    -->
-    <PropertyGroup>
-      <GlobalJsonPath>$([MSBuild]::GetPathOfFileAbove(global.json))</GlobalJsonPath>
```
[the remark this answers is not in the document]
Comment (brandonpollett) [reply test/Microsoft.Health.Dicom.Web.Tests.E2E/Microsoft.Health.Dicom.Web.Tests.E2E.csproj:71]: I believe this also has to do with the variant behavior we've seen between patch versions.

**Review by brandonpollett — APPROVED** [review]
We need to look at any changes needed for the thirdpartynotices file and verify what behavior we're expecting/targeting with the `global.json` changes.

**Pull request merged**